Add "Compute utilization at Mila" page [DOC-473] (#457)
mila-iqia/mila-docs: docs/userguides/compute_utilization_guidelines.md

# Compute utilization at Mila

## Compute powers Mila’s success
Compute is central to most research projects at Mila, and as such, is considered
a key driver of research breakthroughs. Making the absolute most of the available
capacity, i.e. optimal compute utilization, is vital to the success of Mila and
all its researchers.

Yet, a portion of the internal compute power allocated to Mila remains 
inefficiently used (inefficient jobs leave GPU unused) which slows down
research progress. Therefore, raising awareness around efficient compute usage

> has become key, so everyone may run more experiments and get results faster.

Reviewer: I personally feel like this could be tighter and more direct.

Reviewer 2: Ok to replace 1st sentence by : 
"Compute is central to most research at Mila and a key driver of scientific breakthroughs."

After this we should be good to go, correct ? @soline-b and @lebrice ?

Reviewer 3: I let Fabrice be judge on this one :)
But yes: I prefer it with the modification

## What we expect from you
As a researcher, you are expected to request the right amount of compute for
your experiments and use allocated resources efficiently. This ensures that
your experiments do not negatively impact fellow researchers who also need
compute resources.

## The “Safety net” system
To ensure everyone benefits from the compute power made available to the

> scientific community, the [IDT (Innovation, Development and Technologies) team](../home/teams.md)

Reviewer: ```suggestion
scientific community, the [IDT (Innovation, Development and Technology) team](../home/teams.md)
```

has introduced an awareness system designed to prevent unused GPU by empowering

> users with adequate tools.

Reviewer: ```suggestion
has introduced an awareness system designed to prevent unused GPUs by giving
users adequate tools to identify and prevent compute waste.
```

> We monitor cluster performance metrics, focusing strictly on hardware telemetry,

Reviewer: Nitpick, but I'd either use "We" (as if we are speaking to the reader), or "IDT" (as if the docs is neutral), not a mix of both.

never private data or source code. This enables us to provide actionable usage

> reports and notifications, and help you eliminate compute waste.

Reviewer: ```suggestion
never private data or source code. This enables us to provide usage
reports and notifications, and help you eliminate compute waste.
```

| Tool { style="width: 25%" }   | Purpose { style="width: 25%" }    | What’s in it for you { style="width: 50%" }  |
| ---------                     | -------------                     | --------------------  |
| Reports & Dashboard           | ⏱️ Continuous tracking            | Access your data via the public dashboard. Monthly reports sent via Slack to help you track your usage. |
| Waste notification            | 🔔 Early warning                  | Stay informed in case of significantly low usage, receive tips, and get in touch whenever needed. |
| IDT Interventions             | 👨‍💻 Expert coaching                | Benefit from 1-on-1 coaching with an expert to analyze your scripts and optimize job efficiency.  |

## How IDT may help you
We understand that right-sizing the compute you require and using it optimally

> can be complex. IDT is here to support you with:

Reviewer: ```suggestion
We understand that it can be difficult to estimate how much compute your experiments will actually need.
IDT is here to support you with:
```

- One-on-one expert guidance during [Office Hours](../help/office_hours.md) or via [Slack](../help/index.md).
- Support with tracking and reporting tools to help you understand your compute usage.
- Standard resources and guides to help you diagnose and improve.

> This system has been co-created with the Mila Research Compute Committee and

Reviewer 3: Don't we also mention IT team?

will be improved over time. Thank you in advance for sharing your feedback and

> suggestions at research-computing-committee@mila.quebec.

Reviewer: Do not include the internal committee email here. Point instead to an intranet page with the committee description. (Sorry I dont have the URL in hand)
